### PR TITLE
Fix hc sandbox dpki options

### PR DESCRIPTION
**Pull request by jost-s**
### Summary
HC sandbox: Fix `--no-dpki` option which previously enabled DPKI in the conductor when set, instead of disabling it. That lead to `--dpki-network-seed` also being malfunctional. Tests added for `--no-dpki` and `--dpki-network-seed`.

### TODO:
- [x] CHANGELOGs updated with appropriate info
- [x] All code changes are reflected in docs, including module-level docs

### CODE REVIEW

File: crates/hc_sandbox/src/cmds.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ pub struct Create {
     pub in_process_lair: bool,
 
     /// Launch Holochain with the DPKI service disabled.
-    #[arg(long, action = ArgAction::SetFalse)]
```
Comment (ThetaSinner) [inline crates/hc_sandbox/src/cmds.rs:44]: That is seriously confusing...

I see I reviewed that! My mistake.

File: crates/hc_sandbox/tests/cli.rs [hunk 2]
```diff
@@ -276,6 +279,97 @@ async fn generate_sandbox_memproof_deferred_and_call_list_dna() {
...
+        .arg("--piped")
+        .arg("create")
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .stderr(Stdio::inherit())
+        .kill_on_drop(true);
+
+    let mut sandbox_process = input_piped_password(&mut cmd).await;
+    let conductor_config = get_created_conductor_config(&mut sandbox_process).await;
+    assert_eq!(conductor_config.dpki.no_dpki, false);
```
Comment (ThetaSinner) [inline crates/hc_sandbox/tests/cli.rs:300]:
```suggestion
    assert!(!conductor_config.dpki.no_dpki);
```

Or is that equally confusing as the above? :)

File: crates/hc_sandbox/tests/cli.rs [hunk 3]
```diff
@@ -276,6 +279,97 @@ async fn generate_sandbox_memproof_deferred_and_call_list_dna() {
     assert!(exit_code.success());
 }
 
+/// Create a new default sandbox which should have DPKI enabled in the conductor config.
+#[tokio::test(flavor = "multi_thread")]
+async fn default_sandbox_has_dpki_enabled() {
```
Comment (ThetaSinner) [inline crates/hc_sandbox/tests/cli.rs:284]: Nice test!

**Review by ThetaSinner — APPROVED** [review]

**Comment by ThetaSinner** [comment]
Looks like the tests aren't quite there yet? The concept looks good though. Happy to re-review when the tests are ready

File: crates/hc_sandbox/tests/cli.rs (hunk 2 above)
Comment (jost-s) [reply crates/hc_sandbox/tests/cli.rs:300]: It's similar, but clippy enforces how you suggested.

File: crates/hc_sandbox/tests/cli.rs [hunk 4]
```diff
@@ -276,6 +279,97 @@ async fn generate_sandbox_memproof_deferred_and_call_list_dna() {
...
+        .arg("create")
+        .arg("--no-dpki")
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .stderr(Stdio::inherit())
+        .kill_on_drop(true);
+
+    let mut sandbox_process = input_piped_password(&mut cmd).await;
+    let conductor_config = get_created_conductor_config(&mut sandbox_process).await;
+    assert_eq!(conductor_config.dpki.no_dpki, true);
```
Comment (cdunster) [inline crates/hc_sandbox/tests/cli.rs:322]: I'm surprised that clippy doesn't suggest this as above:
```suggestion
    assert!(conductor_config.dpki.no_dpki);
```

**Review by cdunster — APPROVED** [review]
Small suggestion.

File: crates/hc_sandbox/tests/cli.rs (hunk 4 above)
Comment (cdunster) [reply crates/hc_sandbox/tests/cli.rs:322]: Oh, it does :-)

**Pull request merged**